### ISSUE TITLE
poetry should fail when it can't upload a dist

### ISSUE THREAD

**Issue by cjw296** [python-poetry/poetry#920]
<!--
  Hi there! Thank you for discovering and submitting an issue.

  Before you submit this; let's make sure of a few things.
  Please make sure the following boxes are ticked if they are correct.
  If not, please try and fulfill these first.
-->

<!-- Checked checkbox should look like this: [x] -->
- [X] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [X] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: See below
- **Poetry version**:  See below

## Issue

See this build log:
https://circleci.com/gh/cjw296/carthorse/26

Even though it failed to upload the wheel, poetry returns a zero return code.

I can reproduce this locally too:

```
$ poetry publish --build
Building carthorse (0.1.0)
 - Building sdist
 - Built carthorse-0.1.0.tar.gz

 - Building wheel
 - Built carthorse-0.1.0-py2.py3-none-any.whl

Publishing carthorse (0.1.0) to PyPI
 - Uploading carthorse-0.1.0-py2.py3-none-any.whl 100%
                                                                                                                             
[HTTPError]                                                                                                   
400 Client Error: File already exists. See https://pypi.org/help/#file-name-reuse for url: https://upload.pypi.org/legacy  
/                                                                                                                          
                                                                                                                             
publish [-r|--repository REPOSITORY] [-u|--username USERNAME] [-p|--password PASSWORD] [--build]

$ echo $?
0
```

**Comment by madig** [comment]
I ran into the same issue with misspelled upload credentials -- poetry quietly stopped uploading and returned a zero exit status.

**Comment by cjw296** [comment]
@madig - did it write an error to stderr in your case?

**Comment by madig** [comment]
Nothing.

```
> poetry config http-basic.mypypi assadas asdddfdfdsf
> poetry publish -r mypypi
Publishing statmake (0.1.0) to mypypi
> echo $LASTEXITCODE
0
```

**Comment by cjw296** [comment]
Oh, that's even sadder :'(

**Comment by madig** [comment]
Issue has been fixed in current version I think.

**Comment by cjw296** [comment]
@bitprophet has seen this with the latest release attempting to upload to a devpi server. Not error, nothing logged, but also nothing uploaded.

**Comment by bitprophet** [comment]
FWIW that was on Poetry 0.12.17 and I gather there have been newer prereleases floating around. Happy to try again later on something newer.

**Comment by cjw296** [comment]
try this:
```
curl -sSL https://raw.githubusercontent.com/sdispater/poetry/master/get-poetry.py > get-poetry.py
python3.6 get-poetry.py -p
source $HOME/.poetry/env
```

**Comment by bitprophet** [comment]
Same behavior under 1.0.0b8/master unfortunately (though, with less whitespace between the username and password prompts).

Which reminds me we haven't described the full behavior yet: it's not only that there is no upload, but it does prompt for the index's auth; however it's clearly not submitting the auth either as I can put any random garbage in and get no error or other output.

Under 0.12.17, there's maybe 4 lines of whitespace between the username and password prompts; under 1.0.0b8 there are no blank lines. No idea if that helps :joy:

**Comment by jonathanunderwood** [comment]
Issue #2254 is also related to this.

**Comment by dimbleby** [comment]
all seems to be fixed now: I tried with
- a non-existent package
- bad credentials

and got non-zero error codes.  

The redirect case was fixed at #3083.

I think that covers everything in this thread: close it out, and invite folk to raise new issues if they need to.
